controllers/opensearch_reconciler.go

```diff
@@ -839,6 +847,43 @@ func (r OpenSearchReconciler) getRoleMappingListFromSecret(secretName string) ([
 	return mappings, nil
 }
 
+func (r OpenSearchReconciler) getRoleMapping(restClient *util.RestClient, roleName string) (OpenSearchRoleMapping, error) {
+	requestPath := fmt.Sprintf("_plugins/_security/api/rolesmapping/%s", roleName)
+	_, responseBody, err := restClient.SendRequest(http.MethodGet, requestPath, nil)
+	if err != nil {
+		log.Error(err, "An error occurred during audit config request")
+		return OpenSearchRoleMapping{}, err
+	}
+	var mappings OpenSearchRoleMapping
+	if err = json.Unmarshal(responseBody, &mappings); err != nil {
+		return OpenSearchRoleMapping{}, err
+	}
+	return mappings, nil
+}
+
+func (r OpenSearchReconciler) UpdateRoles(restClient *util.RestClient, userName string, role string) error {
+	log.Info(fmt.Sprintf("start Update mapping roles for user:%s ", userName))
+	requestPath := "_plugins/_security/api/rolesmapping"
+	body := fmt.Sprintf(`[{"op": "add", "path": "/%s", "value": {"backend_roles": ["admin"]}}]`, role)
+	statusCode, responseBody, err := restClient.SendRequest(http.MethodPatch, requestPath, strings.NewReader(body))
+	if err == nil {
+		if statusCode == http.StatusOK {
+			requestPathOpensistro := fmt.Sprintf("_plugins/_security/api/internalusers/%s", userName)
+			bodyOpendistro := `[{"op": "replace", "path": "/opendistro_security_roles", "value": []}]`
+			statusCodeOD, responseBodyOD, err := restClient.SendRequest(http.MethodPatch, requestPathOpensistro, strings.NewReader(bodyOpendistro))
+			if err == nil {
+				if statusCodeOD == http.StatusOK {
+					r.logger.Info("The roles for user is successfully updated")
+					return nil
+				}
+				return fmt.Errorf("opendistro security roles update went wrong: [%d] %s", statusCodeOD, responseBodyOD)
+			}
+		}
+		return fmt.Errorf("mapping roles update went wrong: [%d] %s", statusCode, responseBody)
+	}
+	return nil
```

Comment:
Maybe we need to return `err` here?

Reply:
For what? You mean return err instead of return nil ?